codalab/apps/newsletter/tests/test_newsletter_opt_in.py

```diff
@@ -0,0 +1,133 @@
+import hashlib
+import json
+
+import requests
+from django.test import TestCase
+from django.contrib.auth import get_user_model
+
+from apps.newsletter.models import NewsletterUser
+from codalab import settings
+
+User = get_user_model()
+
+
+class NewsletterOptIn(TestCase):
+    def setUp(self):
+        self.user = User.objects.create(email='test@user.com', username='testuser')
+        self.other_user = User.objects.create(email='other@user.com', username='otheruser')
+        self.newsletter_user = NewsletterUser.objects.create(email='test@user.com')
+
+    def tearDown(self):
+        user_hash = hashlib.md5(self.user.email)
+        get_status = requests.get(
+            settings.MAILCHIMP_MEMBERS_ENDPOINT_NEWSLETTER + '/' + user_hash.hexdigest(),
+            auth=("", settings.MAILCHIMP_API_KEY),
+        )
+
+        if get_status.ok:
+            delete_status = requests.delete(
+                settings.MAILCHIMP_MEMBERS_ENDPOINT_NEWSLETTER + '/' + user_hash.hexdigest(),
+                auth=("", settings.MAILCHIMP_API_KEY),
+            )
+
+            if delete_status.ok:
+                print('Test user has been deleted from the Mailchimp mailing list')
+            else:
+                print('Could not delete this user from the Mailchimp mailing list')
+        else:
+            print('User was not found on the Mailchimp mailing list')
+
+    def test_create_user_newsletter_opt_in_true(self):
```

Comment:
All these tests need better names so we know what each one is supposed to be testing. Like
```python
def test_user_creation_makes_newsletter_user_on_opt_in(self):
```
or something like that

Reply:
renamed tests to be more clear